### PR TITLE
fix for pthread_join() crash in some platforms

### PR DESCRIPTION
**Pull request by madhuy**

### CODE REVIEW

File: src/components/utils/src/threads/posix_thread.cc [hunk 1]
```diff
@@ -272,7 +272,11 @@ Thread::~Thread() {
   finalized_ = true;
   stopped_ = true;
   join();
-  pthread_join(handle_, NULL);
+  //in some platforms pthread_join behaviour is undefined when thread is not created(pthread_create) and call pthread_join.
+  if(handle_) {
+    pthread_join(handle_, NULL);
+    handle_ = 0;
```
Comment (anosach-luxoft) [inline src/components/utils/src/threads/posix_thread.cc]: @madhuy this assignment is redundant

File: src/components/utils/src/threads/posix_thread.cc [hunk 2]
```diff
@@ -272,7 +272,11 @@ Thread::~Thread() {
   finalized_ = true;
   stopped_ = true;
   join();
-  pthread_join(handle_, NULL);
+  //in some platforms pthread_join behaviour is undefined when thread is not created(pthread_create) and call pthread_join.
```
Comment (anosach-luxoft) [inline src/components/utils/src/threads/posix_thread.cc]: @madhuy please follow google coding style and 80 symbols limit

File: src/components/utils/src/threads/posix_thread.cc [hunk 3]
```diff
@@ -272,7 +272,11 @@ Thread::~Thread() {
...
+  if(handle_) {
+    pthread_join(handle_, NULL);
```
Comment (anosach-luxoft) [inline src/components/utils/src/threads/posix_thread.cc]: @madhuy BTW you can use some kind of `DCHECK` here

Comment (madhuy) [inline src/components/utils/src/threads/posix_thread.cc]: I don't want assert . i want skip joining thread when it is not created.

**Comment by justinjdickow** [comment]
@AGaliuzov Can you comment on the status of this PR?

**Comment by anosach-luxoft** [comment]
@justinjdickow, @madhuy Changes are ok but is it normal to merge it to `master` branch not to `develop`?

**Comment by justinjdickow** [comment]
It's a hottie so it should be merged into master (the bug exists in master)

**Comment by madhuy** [comment]
@AGaliuzov Could you please merge if you dont have any comments?

**Comment by justinjdickow** [comment]
@AGaliuzov please review

**Comment by LuxoftAKutsan** [comment]
@madhuy would be nice to provide steps to reproduce crash without changes. Or create Unit tests that will fails without changes and pass with them.

**Comment by madhuy** [comment]
@LuxoftAKutsan 
here is issue which has more details  https://github.com/smartdevicelink/sdl_core/issues/452
It is reproducible with some of linux(ARM) machines not all linux

**Comment by LuxoftAKutsan** [comment]
@madhuy OK. But any way unit test for this case would be very useful to avoid such situation in future.

**Pull request merged**